pallets/creditcoin/src/tests.rs

```diff
@@ -3236,3 +3237,49 @@ fn register_address_v2_should_error_with_unsupported_blockchain() {
 		);
 	});
 }
+
+#[test]
+fn request_collect_coins_gate_should_error_when_faucet_not_set() {
+	ExtBuilder::default().build_and_execute(|| {
+		System::set_block_number(1);
+
+		let who = AccountId::new([0; 32]);
+
+		let external_addr = ExternalAddress::default();
+		let tx_id = ExternalTxId::default();
+		let who = Origin::signed(who);
+
+		let contract = TokenContract::GATE(external_addr, tx_id);
+
+		assert_noop!(
+			Creditcoin::request_collect_coins_v2(who, contract),
+			crate::Error::<Test>::BurnGATEFaucetNotSet
+		);
+	});
+}
+
+#[test]
+fn set_burn_gate_contract_fails_with_non_root() {
+	ExtBuilder::default().build_and_execute(|| {
+		let acct: AccountId = AccountId::new([0; 32]);
+		let gate_contract = DeployedContract::default();
+
+		assert_noop!(
+			Creditcoin::set_burn_gate_contract(Origin::signed(acct), gate_contract),
+			BadOrigin
+		);
+	});
+}
+
+#[test]
+fn set_burn_gate_faucet_address_fails_with_non_root() {
+	ExtBuilder::default().build_and_execute(|| {
+		let acct: AccountId = AccountId::new([0; 32]);
+		let addr: AccountId = AccountId::new([0; 32]);
+
+		assert_noop!(
+			Creditcoin::set_burn_gate_faucet_address(Origin::signed(acct), addr),
```

Comment:
```suggestion
			Creditcoin::set_burn_gate_faucet_address(Origin::signed(acct.clone()), acct),
```

and you can then remove `addr`